graphql/queryResolvers/moduleCrud.ts

```diff
@@ -0,0 +1,44 @@
+import type { MutationDeleteModuleArgs, MutationAddModuleArgs } from '..'
+import prisma from '../../prisma'
+import { Context } from '../../@types/helpers'
+import { isAdminOrThrow } from '../../helpers/isAdmin'
+import type { Module } from '@prisma/client'
+
+export const modules = (): Promise<Module[]> => {
+  return prisma.module.findMany({
+    include: {
+      author: true,
+      lesson: true
+    }
+  })
+}
+
+export const addModule = async (
+  _parent: void,
+  args: MutationAddModuleArgs,
+  ctx: Context
+): Promise<Module> => {
+  const { req } = ctx
+  await isAdminOrThrow(req)
+  const { content, lessonId, name } = args
+  const authorId = req.user?.id
+  if (!authorId) throw new Error('No User')
+  const module = await prisma.module.create({
+    data: { authorId, content, lessonId, name }
+  })
+  return module
+}
+
+export const deleteModule = async (
+  _parent: void,
+  arg: MutationDeleteModuleArgs,
+  ctx: Context
+) => {
```

Comment:
Looks like this function is missing typescript definition for what it returns

Reply:
This is a resolver function so it may be redundant to write typescript return value. Also, the return value is already defined in the typedef. 

None of the other resolvers have output types: https://github.com/garageScript/c0d3-app/blob/master/graphql/queryResolvers/allUsers.ts

Reply:
From my perspective, it makes the function more readable for me when defined with a typescript definition. When I look at the function, I can see exactly what it is returning without having to look into the function's code or without having to refer back to typedefs. Missing typescript definition for what the function returns means we aren't taking advantage of typescript to be more clear in our code. 

Just because we have been doing things this way(`not having typescript return value on resolvers`) doesn't mean its best IMO. For example, no files currently use any kind of `constants file` to pull constant values from, and although we have been doing things this way, I think its way better to make a constants file rather than not have one.

Reply:
> it makes the function more readable for me when defined with a typescript definition. When I look at the function, I can see exactly what it is returning without having to look into the function's code

100% agree for functions.

> I can see exactly what it is returning without having to look into the function's code or without having to refer back to typedefs.

For resolvers though, it is better to always refer to typedefs as the source of truth, because it is the source of truth. If you have typedefs AND define a return type for the function then you have additional work or making sure they are in sync and potentially run the risk of making them out of sync (therefore errors).

For the reasons above, it is better (and safer) to not have return types in resolvers.